File: backend/CRUD/Services/AddressService.cs
using CRUD.DTO;
using CRUD.Repositories.Interfaces;
using CRUD.Services.Interfaces;
using CRUD.Entities;

namespace CRUD.Services
{
    public class AddressService : IAddressService
    {
        private readonly IAddressRepository addressRepository;
        private readonly HttpClient httpClient;

        public AddressService(IAddressRepository addressRepository, HttpClient httpClient)
        {
            this.addressRepository = addressRepository;
            this.httpClient = httpClient;
        }

        public async Task<ViaCEPResponse> GetAddressByCEP(int CEP)
        {
            var url = $"https://viacep.com.br/ws/{CEP}/json/";
            var response = await httpClient.GetAsync(url);

            var result = await response.Content.ReadFromJsonAsync<ViaCEPResponse>();
            if(result == null)
            {
                throw new Exception("Error getting address");
            }
            return result;
      
        }

        public async Task CreateAddress(AddressDTO request, int personId)
        {
            var address = new Address
            {
                PersonId = personId,
                Street = request.Street,
                City = request.City,
                Number = request.Number,
                Neighborhood = request.Neighborhood,
                CEP = request.CEP,
                AddressType = request.addressType,
            };
            await addressRepository.CreateAddress(address);
        }


        public async Task DeleteAddress(int personId, int addressId)
        {
            await addressRepository.DeleteAddress(personId, addressId);
        }
    }
}

File: frontend/src/app/services/address-service/address.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { AddressDTO } from '../../models/DTO/addressDTO';
import { ViaCEPResponse } from '../../models/viaCEPresponse';

@Injectable({
  providedIn: 'root'
})
export class AddressService {
  
  public constructor(private http: HttpClient){}

  private url = 'https://localhost:7234/api/Address';

  public async PostAddress(addressDTO : AddressDTO, personId: number) : Promise<void>{
    try{
      await this.http.post(`${this.url}/${personId}`, addressDTO).toPromise();
    }
    catch{
      throw new Error;
    }
  }

  public async DeleteAddress(personId: number, addressId: number) : Promise<void>{
    try{
      await this.http.delete(`${this.url}/${personId}/${addressId}`).toPromise();
    }
    catch{
      throw new Error;
    }
  }

  public async GetAddressByCEP(cep : number ) : Promise<ViaCEPResponse> {
    try{
      var response =  await this.http.get<ViaCEPResponse>(`${this.url}/${cep}`).toPromise();
      if(response == undefined){
        throw new Error;
      }
      return response;
    }
    catch{
      throw new Error;
    }
  }

}
